front/src/common/Map/Layers/InfraObjectLayers/SpeedLimits.tsx

```diff
@@ -2,22 +2,24 @@ import { isNil } from 'lodash';
 import type { ExpressionSpecification, FilterSpecification } from 'maplibre-gl';
 import { Source } from 'react-map-gl/maplibre';
 import type { SymbolLayer, LineLayer } from 'react-map-gl/maplibre';
-import { useSelector } from 'react-redux';
 
 import { MAP_URL } from 'common/Map/const';
 import OrderedLayer from 'common/Map/Layers/OrderedLayer';
-import type { RootState } from 'reducers';
 import type { MapState } from 'reducers/map';
+import type { LayersSettings } from 'reducers/osrdconf/types';
 import type { Theme, OmitLayer } from 'types';
 
 interface SpeedLimitsProps {
   colors: Theme;
   layerOrder: number;
   punctualLayerOrder: number;
   infraID?: number;
+  layersSettings: MapState['layersSettings'] | LayersSettings;
```

Comment:
You can change MapState type so layersSettings is a LayerSettings
```suggestion
  layersSettings: MapState['layersSettings'];
```